orchestra/workflow/load.py

```diff
@@ -0,0 +1,161 @@
+
+    if not version_created:
+        if not force:
+            raise WorkflowError('Version {} already exists'
+                                .format(version_data['slug']))
+
+        # Check that the versions are safe to merge
+        new_step_slugs = set(s['slug'] for s in version_data['steps'])
+        old_step_slugs = set(
+            WorkflowStep.objects
+            .filter(workflow_version=version)
+            .values_list('slug', flat=True)
+        )
+        if new_step_slugs != old_step_slugs:
+            raise WorkflowError('Even with --force, cannot change the steps '
+                                'of a workflow. Drop and recreate the '
+                                'database to reset, or create a new version '
+                                'for your workflow.')
+
+    # Create or update the version steps.
+    for step_data in version_data['steps']:
+        step, step_created = create_or_update(
+            WorkflowStep,
+            {
+                'slug': step_data['slug'],
+                'workflow_version': version
+            },
+            {
+                'name': step_data['name'],
+                'description': step_data['description'],
+                'is_machine': step_data.get('is_machine', False),
+                'execution_function': step_data.get('execution_function', {}),
+                'assignment_policy': step_data.get('assignment_policy', {}),
+                'review_policy': step_data.get('review_policy', {}),
+                'user_interface': step_data.get('user_interface', {}),
+            }
+        )
+
+        # TODO(dhaas): should we prevent updates to these?
+        required_certification_slugs = step_data.get(
+            'required_certifications', [])
+        required_certifications = list(Certification.objects.filter(
+            workflow=workflow,
+            slug__in=required_certification_slugs
+        ))
+        step.required_certifications = required_certifications
+
+    # Set up step dependencies once the steps objects are in the DB.
+    # TODO(dhaas): should we prevent updates to these?
+    for step_data in version_data['steps']:
+        step = WorkflowStep.objects.get(
+            slug=step_data['slug'],
+            workflow_version=version
+        )
+        creation_depends_on_slugs = step_data.get('creation_depends_on', [])
+        creation_depends_on = WorkflowStep.objects.filter(
+            workflow_version=version,
+            slug__in=creation_depends_on_slugs
+        )
+        step.creation_depends_on = creation_depends_on
+        submission_depends_on_slugs = step_data.get('submission_depends_on',
+                                                    [])
+        submission_depends_on = WorkflowStep.objects.filter(
+            workflow_version=version,
+            slug__in=submission_depends_on_slugs
+        )
+        step.submission_depends_on = submission_depends_on
+
+
+def create_or_update(model, create_keys, update_keys):
```

Comment:
seems like a common bit of functionality that we should surface, perhaps in `https://github.com/unlimitedlabs/orchestra/tree/master/orchestra/utils`

`utils.django`?